### PR TITLE
working DNS status feature

### PR DESCRIPTION
**Pull request by lazypower**
Still a candidate for refactoring, but this greenlights the deployment w/
    SkyDNS powering local service discovery

### CODE REVIEW

**Comment by lazypower** [comment]
@mbruzek dont review this yet, its apparently got merge issues. I want to refactor this a bit and will rebase on top of master.

Monday AM should be sweet :)

**Comment by mbruzek** [comment]
Thanks for the comment.  I pulled this up for review today, but will defer until I hear from you.

**Comment by lazypower** [comment]
@mbruzek - i rebased. this should be g2g - use a cautious eye to make sure i didn't override something critical. this passes deploy / validation tests here w/ the shipped guestbook action

juju action do k8s/0 guestbook-example will turn up the guestbook sample.

kubectl get svc
curl 10.1.x.x 

Should be g2g once all the pods are online. this takes ~ 15 seconds from request to completion.

File: reactive/k8s.py [hunk 1]
```diff
@@ -31,11 +32,29 @@ def config_changed():
         docker_compose_kill_remove('proxy')
         remove_state('proxy.available')
 
+@when('kubelet.available', 'proxy.available', 'cadvisor.available')
+def final_messaging():
+    ''' Lower layers emit messages, and if we do not clear the status messaging
+        queue, we are left with whatever the last method call sets status to.
         if config.changed('version'):
```
Comment (mbruzek) [inline reactive/k8s.py]: This if statement looks out of place in this comment, unless you are specifically commenting on that.

File: reactive/k8s.py [hunk 2]
```diff
@@ -31,11 +32,31 @@ def config_changed():
         docker_compose_kill_remove('proxy')
         remove_state('proxy.available')
 
-        if config.changed('version'):
-            hookenv.log('Removing kubectl.downloaded state so the new version'
-                        ' of kubectl will be downloaded.')
-            remove_state('kubectl.downloaded')
+    if config.changed('version'):
```
Comment (mbruzek) [inline reactive/k8s.py:35]: the level of this if statement changed but this new level looks fine and will always evaluate.

File: reactive/k8s.py [hunk 3]
```diff
@@ -171,3 +212,21 @@ def docker_compose_kill_remove(service):
...
+        status_set('maintenance', 'Starting the kubernetes master container')
+        # Start the kubelet container that starts the three master services.
+        check_call(split('docker-compose up -d kubelet'))
+        set_state('kubelet.available')
+        # Start the proxy container
+        status_set('maintenance', 'Starting the kubernetes proxy container')
+        check_call(split('docker-compose up -d proxy'))
+        set_state('proxy.available')
+
+def recycle_kubernetes():
```
Comment (mbruzek) [inline reactive/k8s.py:228]: I don't see where you call this method yet.  The point of the config-changed logic was to use the states to properly restart the services by removing the correct states.

**Comment by mbruzek** [comment]
+1 LGTM.  The latest version of this code looks OK.  One method not used but we can remove it or use it later.

**Comment by lazypower** [comment]
We can gut this method matt. I like your approach to recycling the services. I used this for a convenience call when devving and it made its way into this PR :(

**Pull request merged**